File: examples/infolder/backend/go/models/product.go
package models

type Product struct {
	Id string `gorm:"primaryKey;column:id" json:"id"`
	Name string `gorm:"column:name;not null" json:"name"`
	Price float64 `gorm:"column:price;not null" json:"price"`
	Stock int64 `gorm:"column:stock;not null;default:0" json:"stock"`
	Type string `gorm:"column:type;not null;default:physical" json:"type"`
}

File: examples/infolder/lib/models/product/product.dart
import 'package:schemix/schemix.dart';

part 'product.schemix.dart';

@Schemix(
  name: 'products',
  schemaVersion: 1,
)
class Product {
  @PrimaryKey(autoGenerate: true)
  final String id;

  @Length(max: 150)
  final String name;

  @Precision(precision: 10, scale: 2)
  @Min(0.0)
  final double price;

  @Min(0)
  @Max(10000)
  @DatabaseDefault(0)
  final int stock;

  @AllowedValues(['physical', 'digital'])
  @DatabaseDefault('physical')
  final String type;

  const Product({
    required this.id,
    required this.name,
    required this.price,
    this.stock = 0,
    this.type = 'physical',
  });

  factory Product.fromJson(Map<String, dynamic> json) => _$ProductFromJson(json);
  Map<String, dynamic> toJson() => _$ProductToJson(this);
  Product copy() => _$ProductCopy(this);
}
